cider/exec/nextgen/operators/RowToColumnNode.cpp

```diff
@@ -55,13 +55,77 @@ class ColumnWriter {
       case kDOUBLE:
         writeFixSizedTypeCol();
         break;
+      case kVARCHAR:
+      case kCHAR:
+      case kTEXT:
+        writeVariableSizeTypeCol();
+        break;
       default:
         LOG(FATAL) << "Unsupported data type in ColumnWriter: "
                    << expr_->get_type_info().get_type_name();
     }
   }
 
  private:
+  void writeVariableSizeTypeCol() {
+    // Get values need to write
+    utils::VarSizeJITExprValue values(expr_->get_expr_value());
+
+    // Allocate buffer
+
+    // the null_buffer, raw_data_buffer not used anymore.
+    // so it doesn't matter whether null_buffer is nullptr
+    // or constant false.
+    auto null_buffer = JITValuePointer(nullptr);
+    if (!expr_->get_type_info().get_notnull()) {
+      // TBD: Null representation, bit-array or bool-array.
+      null_buffer.replace(context_.getJITFunction()->createLocalJITValue(
+          [this]() { return allocateBitwiseBuffer(0); }));
+
+      context_.getJITFunction()->emitRuntimeFunctionCall(
+          "set_null_vector_bit",
+          JITFunctionEmitDescriptor{
+              .ret_type = JITTypeTag::VOID,
+              .params_vector = {
+                  {null_buffer.get(), index_.get(), values.getNull().get()}}});
+    }
+
+    // offset, need array_len + 1 element.
+    auto raw_length_buffer = context_.getJITFunction()->createLocalJITValue([this]() {
+      auto bytes = (arrow_array_len_ + 1) *
+                   context_.getJITFunction()->createLiteral(JITTypeTag::INT64, 4);
+      // TODO: should set memory to 0.
```

Comment:
It eventually call `std::allocator.allocate()`, will still need set memory to 0?